### PR TITLE
Upgrade hikari 11249

### LINKED ISSUE #11249: upgrade hikariCP to jdk8 for proxy
Hi,All.
We need tp upgrade the hikari to jdk8 since some feature will depend on it. Such as metics in prometheus. It also can avoid some uncertain class load problem for coexistance of hikariCP-java7 & hikari.

### PR DESCRIPTION
**Pull request by dachuan9e**
Fixes #11249.

Changes proposed in this pull request:
- remove the HikariCP-java7, only use HikariCP for scaling and proxy.

### CODE REVIEW

File: shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-postgresql/pom.xml [hunk 1]
```diff
@@ -34,7 +34,11 @@
             <artifactId>shardingsphere-scaling-core</artifactId>
             <version>${project.version}</version>
         </dependency>
-        
+        <dependency>
+            <groupId>com.zaxxer</groupId>
+            <artifactId>HikariCP</artifactId>
+            <scope>compile</scope>
```
Comment (TeslaCN) [inline shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-postgresql/pom.xml:40]: `compile` is the default value. Is it necessary?

Comment (dachuan9e) [reply shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-postgresql/pom.xml:40]: It needs during the compilation. Before it use the HikariCP-java7 in elasticjob-lite-core indirectly. 
Meanwhile root pom.xml in ShardingSphere has:
            <dependency>
                <groupId>com.zaxxer</groupId>
                <artifactId>HikariCP</artifactId>
                <version>${hikari-cp.version}</version>
                <scope>test</scope>
            </dependency>
The sub module will inherit it.  Therefore scope should be marked compile explicitly.

**Comment by tristaZero** [comment]
Please take care of these failed CI tests.

File: shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-postgresql/pom.xml (hunk 1 above)
Comment (TeslaCN) [reply shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-postgresql/pom.xml:40]: Consider configuring the `dependencyManagement` in the upper pom.

**Review by tristaZero — APPROVED** [review]

**Pull request merged**